### PR TITLE
[ DTS ] MatMul

### PR DESCRIPTION
**Pull request by jane-intel**
MatMul [specification](https://docs.openvinotoolkit.org/latest/_docs_ops_matrix_MatMul_1.html)

### CODE REVIEW

File: inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_matmul.cpp [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+}
+INSTANTIATE_TEST_CASE_P(MatMul, DynamicToStaticShapeMatMul, combinations);
+
+TEST_P(DynamicToStaticShapeMatMulAInputDSR, CompareFunctions) {
+}
+INSTANTIATE_TEST_CASE_P(MatMulSingleDSR, DynamicToStaticShapeMatMulAInputDSR, combinations);
+
+TEST_P(DynamicToStaticShapeMatMulBInputDSR, CompareFunctions) {
+}
+INSTANTIATE_TEST_CASE_P(MatMulSingleDSR, DynamicToStaticShapeMatMulBInputDSR, combinations);
```
Comment (gladilov-gleb) [inline inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_matmul.cpp:410]: What is the difference between 3 test classes? Now it looks like the only difference is in input nodes generation - processing is the same. If so, I think we can get rid of code duplication

File: inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_matmul.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+TEST_P(DSR_MatMulAInput, CompareWithReference) {
+    Run();
+}
+INSTANTIATE_TEST_CASE_P(DISABLED_DynamicMatMul, DSR_MatMulAInput, combinations);
+
+TEST_P(DSR_MatMulBInput, CompareWithReference) {
+    Run();
+}
+INSTANTIATE_TEST_CASE_P(DISABLED_DynamicMatMul, DSR_MatMulBInput, combinations);
```
Comment (gladilov-gleb) [inline inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_matmul.cpp:136]: Why tests are disabled?

File: inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_matmul.cpp (hunk 1 above)
Comment (jane-intel) [reply inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_matmul.cpp]: Agree. I redesigned testing approach for MatMul. Please see if it is what you meant.

File: inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_matmul.cpp (hunk 2 above)
Comment (jane-intel) [reply inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_matmul.cpp]: Enabled them for now. But I have no idea what to expect -- I have no access to my stick for now. Could you please help with error identification in CI, if any?

**Comment by jane-intel** [comment]
@gladilov-gleb, smth really goes wrong. I filed a ticket with the link to failure in CI and disabled inference tests in current PR. Ticket number is placed in the comment near disabled test.

**Comment by gladilov-gleb** [comment]
> @gladilov-gleb, smth really goes wrong. I filed a ticket with the link to failure in CI and disabled inference tests in current PR. Ticket number is placed in the comment near disabled test.

Thanks

**Comment by gladilov-gleb** [comment]
I'm not sure it's okay to use "JIRA" near ticket number..

**Comment by jane-intel** [comment]
> I'm not sure it's okay to use "JIRA" near ticket number..

It is okay. JIRA is well known service for task tracking. 
But project name mentioning and abbreviations are forbidden

**Pull request merged**